### PR TITLE
Add unittests for group filtering (Azure)

### PR DESCRIPTION
**Pull request by pasoroki**

### CODE REVIEW

File: providers/session_state_test.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func TestSessionStateSerialization(t *testing.T) {
 	}
 	encoded, err := s.EncodeSessionState(c)
 	assert.Equal(t, nil, err)
-	assert.Equal(t, 3, strings.Count(encoded, ":"))
+	assert.Equal(t, 4, strings.Count(encoded, ":"))
```
Comment (thenewwazoo) [inline providers/session_state_test.go:29]: Needs a test including a group spec

File: providers/provider_data.go [hunk 2]
```diff
@@ -13,6 +13,7 @@ type ProviderData struct {
 	ProfileURL        *url.URL
 	ProtectedResource *url.URL
 	ValidateURL       *url.URL
+	GroupsURL         *url.URL
```
Comment (thenewwazoo) [inline providers/provider_data.go]: This type of Group API is unique to Azure, isn't it? I don't think it should live in ProviderData if that's the case.

**Review by thenewwazoo — CHANGES_REQUESTED** [review]
Looks really good! I like mocking out the group calls.

File: providers/azure.go [hunk 3]
```diff
@@ -149,11 +150,12 @@ func (p *AzureProvider) GetGroups(s *SessionState, f string) (string, error) {
...
 	//
 	// Filters that will be possible to use:
-	// contains - unknown function | "https://graph.microsoft.com/v1.0/me/memberOf?$filter=contains(displayName,%27olm%27)"
-	// startswith - not supported  | "https://graph.microsoft.com/v1.0/me/memberOf?$filter=startswith(displayName,%27olm%27)"
-	// substring - not supported   | "https://graph.microsoft.com/v1.0/me/memberOf?$filter=substring(displayName,0,2)%20eq%20%27olm%27"
+	// contains - unknown function | "https://graph.microsoft.com/v1.0/me/memberOf?$filter=contains(displayName,%27groupname%27)"
+	// startswith - not supported  | "https://graph.microsoft.com/v1.0/me/memberOf?$filter=startswith(displayName,%27groupname%27)"
+	// substring - not supported   | "https://graph.microsoft.com/v1.0/me/memberOf?$filter=substring(displayName,0,2)%20eq%20%27groupname%27"
+
+	requestUrl := "https://graph.microsoft.com/v1.0/me/memberOf%3F$select=displayName"
```
Comment (bdwyertech) [inline providers/azure.go]: The `%3F` here didn't work for me, swapping back to a `?` seemed to fix it.  That was Ubuntu 16.04 and Go 1.6.2.

File: providers/session_state_test.go (hunk 1 above)
Comment (thenewwazoo) [reply providers/session_state_test.go:29]: Should this be 5 now?

File: providers/azure.go (hunk 3 above)
Comment (thenewwazoo) [reply providers/azure.go]: Hey thanks! Not sure what ate that `?`. @pasoroki, poke this and I'll give it a merge :)

File: providers/session_state_test.go (hunk 1 above)
Comment (pasoroki) [reply providers/session_state_test.go:29]: 5 items in `encoded` string, 4 `:` delimiters. Looks fine to me.
The only thing that is missed here is `PermittedGroups` but they are stored in `AzureProvider` structure and shouldn't be visible in `SessionState` so we're good

File: providers/azure.go (hunk 3 above)
Comment (pasoroki) [reply providers/azure.go]: I confirm, there should be `?` instead of `%3F`

**Review by thenewwazoo — APPROVED** [review]

**Pull request merged**